src/Donors/Repositories/DonorRepository.php

```diff
@@ -427,9 +431,10 @@ public function getDonorLatestDonationDate(int $donorId)
     }
 
     /**
+     * @unreleased change return to DonorType
      * @since 2.20.0
      *
-     * @return string|null
+     * @return DonorType|null
      */
     public function getDonorType(int $donorId)
```

Comment:
I'm thinking this entire method can be replaced with the Donor model.  On another note I feel like the donor type is weird because a donor could be multiple at the same time.

```
// psuedo code

getDonorType(Donor $donor){
  
  if (empty($donor->donations)){
    return DonorType::NEW();
  }
  
  if ((bool)$donor->subscriptions){
    return DonorType::SUBSCRIBER();
  }

  return DonorType::REPEAT();

}
```